numba/tests/test_fancy_indexing.py

```diff
@@ -303,32 +302,43 @@ def np_new_axis_setitem(a, idx, item):
 
 class TestFancyIndexingMultiDim(MemoryLeakMixin, TestCase):
     # Every case has exactly one, one-dimensional array,
-    # otherwise it's not fancy indexing.
+    # Otherwise it's not fancy indexing
     shape = (5, 6, 7, 8, 9, 10)
     indexing_cases = [
         # Slices + Integers
         (slice(4, 5), 3, np.array([0, 1, 3, 4, 2]), 1),
         (3, np.array([0,1,3,4,2]), slice(None), slice(4)),
+        (3, np.array([[0, 1, 3, 4, 2], [0, 1, 2, 3, 2], [3, 1, 3, 4, 1]]),
+         slice(None), slice(4)), # multi-dimensional
 
         # Ellipsis + Integers
         (Ellipsis, 1, np.array([0,1,3,4,2])),
         (np.array([0,1,3,4,2]), 3, Ellipsis),
+        (np.array([[0, 1, 3, 4, 2], [0, 1, 2, 3, 2], [3, 1, 3, 4, 1]]),
+         3, Ellipsis), # multi-dimensional
 
         # Ellipsis + Slices + Integers
         (Ellipsis, 1, np.array([0,1,3,4,2]), 3, slice(1,5)),
         (np.array([0,1,3,4,2]), 3, Ellipsis, slice(1,5)),
+        (np.array([[0, 1, 3, 4, 2], [0, 1, 2, 3, 2], [3, 1, 3, 4, 1]]),
+         3, Ellipsis, slice(1, 5)), # multi-dimensional
 
         # Boolean Arrays + Integers
         (slice(4, 5), 3,
          np.array([True, False, True, False, True, False, False]),
          1),
         (3, np.array([True, False, True, False, True, False]),
          slice(None), slice(4)),
+
+        # Differently ordered arrays
+        # Ellipsis + Slices + Integers
+        (Ellipsis, 1, np.array([0, 1, 3, 4, 2], order='A'), 3, slice(1, 5)),
+        (np.array([0, 1, 3, 4, 2], order='F'), 3, Ellipsis, slice(1, 5)),
+        (np.array([[0, 1, 3, 4, 2], [0, 1, 2, 3, 2], [3, 1, 3, 4, 1]], order='A'),
```

Comment:
Setting `order='A'` in the NumPy constructor won't have the effect of it being `A` ordered within Numba's type system (assuming that was the intent?) Suggest deferring fixing this to a later PR.